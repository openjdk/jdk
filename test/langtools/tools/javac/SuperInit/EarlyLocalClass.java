/*
 * @test /nodynamiccopyright/
 * @bug 8325805
 * @summary Verify local class in early construction context has no outer instance
 * @compile/fail/ref=EarlyLocalClass.out -XDrawDiagnostics EarlyLocalClass.java
 * @enablePreview
 */
public class EarlyLocalClass {
    EarlyLocalClass() {
        class Local {
            void foo() {
                EarlyLocalClass.this.hashCode();    // this should FAIL
            }
        }
        new Local();                                // this is OK
        super();
        new Local();                                // this is OK
    }
}
