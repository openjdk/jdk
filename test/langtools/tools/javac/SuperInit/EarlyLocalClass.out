EarlyLocalClass.java:12:32: compiler.err.cant.ref.before.ctor.called: this
- compiler.note.preview.filename: EarlyLocalClass.java, DEFAULT
- compiler.note.preview.recompile
1 error
